httputil/auth.go

```diff
@@ -91,6 +105,34 @@ func QueryAuth[T any](key string, fn TokenAuthenticatorFunc[T]) AuthenticateFunc
 	}
 }
 
+func BasicAuth[T any](authFn BasicAuthenticatorFunc[T]) AuthenticateFunc[T] {
+	return func(r *http.Request) (T, error) {
+		var empty T
+
+		token := strings.TrimPrefix(r.Header.Get(BasicAuthHeader), BasicAuthPrefix)
+		if token == "" {
+			// Fallback to the proxy header if available
+			token = strings.TrimPrefix(r.Header.Get(BasicAuthProxyHeader), BasicAuthPrefix)
+		}
+
+		if token == "" {
+			return empty, ErrBasicAuthenticate
+		}
+
+		payload, err := base64.StdEncoding.DecodeString(token)
+		if err != nil {
+			return empty, ErrBasicAuthenticate
+		}
+
+		pair := bytes.SplitN(payload, []byte(":"), 2)
+		if len(pair) != 2 {
+			return empty, ErrUnauthorized
```

Comment:
non-blocking

could potentially fmt.Errorf with some more context in some of these error cases for a logging level troubleshooting info.